
 This patch can be applied to version 3.0.1 and should
 fix bug #1307 where a Job being escalated with 
 Allow Higher Duplicates = no is cancelled.

 Apply it to version 3.0.1 with:

 cd <bacula-source>
 patch -p0 <3.0.1-higher-duplicates.patch
 ./configure <your-options>
 make
 ...
 make install



Index: src/dird/job.c
===================================================================
--- src/dird/job.c	(revision 8902)
+++ src/dird/job.c	(working copy)
@@ -672,6 +672,9 @@
    if (!job->AllowHigherDuplicates) {
       foreach_jcr(djcr) {
          char ec1[50];
+         if (jcr == djcr) {
+            continue;                   /* do not cancel this job */
+         }
          if (strcmp(job->name(), djcr->job->name()) == 0) {
             bool cancel_queued = false;
             if (job->DuplicateJobProximity > 0) {
